components/ListShow/ListShow.jsx

```diff
@@ -1,51 +1,97 @@
-import {useParams, Link} from 'react-router-dom'
+import {useParams} from 'react-router-dom'
 import {useState, useEffect} from 'react'
 import * as usersService from '../../services/usersService'
 
-//* /lists/:listId
 const ListShow = (props) => {
 	const {user} = props
 	const {listId} = useParams()
-	const {list, setList} = useState(null)
+	const [list, setList] = useState(null)
+	const [isEditing, setIsEditing] = useState(false)
+	const [editFormData, setEditFormData] = useState(null)
 
 	useEffect(() => {
 		const fetchList = async () => {
-			const userId = user.id
+			const userId = user._id
 			const fetchedList = await usersService.showList(userId, listId)
 			setList(fetchedList)
 		}
 		fetchList()
-	}, [listId, user.id])
+	}, [listId, user._id])
 
-	
 	const handleDeleteListItem = async (itemId) => {
-		const deletedListItem = await usersService.deleteListItem(itemId)
+		const deletedListItem = await usersService.deleteListItem(user,itemId)
 		setList({
 			...list,
-			items: list.items.filter((item) => item.id !== deletedListItem.id),
+			items: list.items.filter((item) => item._id !== deletedListItem._id),
 		})
-    }
-    
-    if(!list) return <p>Can't find your list. Please try again later</p>
+	}
+
+	const handleEditClick = () => {
+		setEditFormData({listName: list.listName})
+		setIsEditing(true)
+	}
+
+	const handleEditChange = (event) => {
+		const inputName = event.target.name
+		const inputValue = event.target.value
+		setEditFormData({
+			...editFormData,
+			[inputName]: inputValue,
```

Comment:
This will only work for list name and not list items. We could probably keep it incase we add new typeable fields later but we need to be careful it doesn't add unwanted data.